lib/routemap.c

```diff
@@ -1757,7 +1779,13 @@ static int route_map_dep_update(struct hash *dephash, const char *dep_name,
 		if (!dep->this_hash)
 			dep->this_hash = dephash;
 
-		hash_get(dep->dep_rmap_hash, rname, route_map_name_hash_alloc);
+		dep_data = hash_lookup(dep->dep_rmap_hash, rname);
```

Comment:
Please create an appropriate `struct route_map_dep_data` and pass it in.  Just passing rname is a crash waiting to happen if someone adds new data to the code.

Reply:
Thanks, @donaldsharp. I've addressed it.